tests/trainer/test_states.py

```diff
@@ -0,0 +1,49 @@
+from pytorch_lightning import Trainer
+from pytorch_lightning.trainer.states import TrainerState
+from tests.base import EvalModelTemplate
+
+
+def test_initialize_state(tmpdir):
+    """
+    Tests that state is INITIALIZE after Trainer creation
+    """
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        fast_dev_run=True,
+    )
+
+    assert trainer.state == TrainerState.INITIALIZE
+
+
+def test_finished_state_after_fit(tmpdir):
```

Comment:
I could not come up with a nice way to test that a state is RUNNING during the fit/test execution. Maybe you have some ideas?

Reply:
you can still test the interrupt...
also, write a simple custom call back which test the status over time

Reply:
Done